File: src/service/CaptchaService.ts
import axios from "axios";

export interface ICaptchaProps {
  host: string;
  siteKey: string;
  captchaUrl: string;
}

export interface IRefProps {
  urlReferencia: string;
  email: string;
  senha: string;
  carteira: string;
}

export interface IReferenciaCarteiraProps {
  carteira: string;
  referencia: string;
}

export default class CaptchaService {
  public async obterCaptcha(dadosCaptcha: ICaptchaProps): Promise<string> {
    const { data } = await axios.request({
      url: `${dadosCaptcha.captchaUrl}/api/v1/captcha/resolver`,
      method: "post",
      headers: {
        "Content-Type": "application/json",
      },
      data: dadosCaptcha,
    });

    return data;
  }

  public async obterReferencias(
    codigoReferencia: string,
    captchaUrl: string
  ): Promise<IReferenciaCarteiraProps[]> {
    const { data } = await axios.request({
      url: `${captchaUrl}/api/v1/referencia/${codigoReferencia}`,
      method: "get",
      headers: {
        "Content-Type": "application/json",
      },
      data: null,
    });

    return data;
  }

  public async enviarUrlReferencia(
    referencia: IRefProps,
    captchaUrl: string
  ): Promise<string> {
    const { data } = await axios.request({
      url: `${captchaUrl}/api/v1/referencia`,
      method: "post",
      headers: {
        "Content-Type": "application/json",
      },
      data: referencia,
    });

    return data;
  }
}

File: android/app/src/main/java/br/com/slimbot/collector/service/CaptchaService.java
package br.com.slimbot.collector.service;

import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.util.Base64;

import br.com.slimbot.collector.vo.CaptchaDataVO;
import br.com.slimbot.collector.vo.CaptchaPropsVO;
import br.com.slimbot.collector.vo.SiteConfigVO;
import br.com.slimbot.collector.vo.TaskVO;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;

public class CaptchaService {

    private OkHttpClient client = new OkHttpClient().newBuilder().build();

    public String resolverCaptcha(CaptchaPropsVO captchaPropsVO) throws IOException {

        MediaType mediaType = MediaType.parse("application/json");

        RequestBody body = RequestBody.create(mediaType,
                String.format("{\"host\":\"%s\",\"siteKey\":\"%s\"}", captchaPropsVO.getHost(), captchaPropsVO.getSiteKey()));

        Request request = new Request.Builder()//
                .url(String.format("%s/api/v1/captcha/resolver",captchaPropsVO.getResolverHost()))//
                .method("POST", body)//
                .addHeader("Content-Type", "application/json").build();

        return client.newCall(request).execute().body().string();
    }

}
